helix-rest/src/main/java/org/apache/helix/rest/server/resources/zookeeper/ZooKeeperAccessor.java

```diff
@@ -0,0 +1,181 @@
+    try {
+      cmd = ZooKeeperCommand.valueOf(commandStr);
+    } catch (Exception e) {
+      return badRequest("Invalid ZooKeeper command: " + commandStr);
+    }
+
+    // Lazily initialize ZkBaseDataAccessor
+    ServerContext _serverContext =
+        (ServerContext) _application.getProperties().get(ContextPropertyKeys.SERVER_CONTEXT.name());
+    _zkBaseDataAccessor = _serverContext.getByteArrayZkBaseDataAccessor();
+
+    // Need to prepend a "/" since JAX-RS regex removes it
+    path = "/" + path;
+    switch (cmd) {
+      case exists:
+        return exists(path);
+      case getBinaryData:
+        return getData(path, cmd);
+      case getStringData:
+        return getData(path, cmd);
+      case getChildren:
+        return getChildren(path);
+      default:
+        LOG.error("Unsupported command :" + commandStr);
+        return badRequest("Unsupported command :" + commandStr);
+    }
+  }
+
+  /**
+   * Checks if a ZNode exists in the given path.
+   * @param path
+   * @return true if a ZNode exists, false otherwise
+   */
+  private Response exists(String path) {
+    if (!isPathValid(path)) {
+      String errMsg = "exists(): The given path {} is not a valid ZooKeeper path!" + path;
+      LOG.error(errMsg);
+      return badRequest(errMsg);
+    }
+
+    Map<String, Boolean> result = ImmutableMap.of(ZooKeeperCommand.exists.name(),
+        _zkBaseDataAccessor.exists(path, AccessOption.PERSISTENT));
+    return JSONRepresentation(result);
+  }
+
+  /**
+   * Reads the given path from ZooKeeper and returns the binary data for the ZNode.
+   * @param path
+   * @param command denotes whether return type should be binary or String
+   * @return binary data in the ZNode
+   */
+  private Response getData(String path, ZooKeeperCommand command) {
+    if (!isPathValid(path)) {
+      String errMsg = "getData(): The given path {} is not a valid ZooKeeper path!" + path;
+      LOG.error(errMsg);
+      return badRequest(errMsg);
+    }
+
+    if (_zkBaseDataAccessor.exists(path, AccessOption.PERSISTENT)) {
+      byte[] bytes = _zkBaseDataAccessor.get(path, null, AccessOption.PERSISTENT);
+      String s = new String(bytes);
+
+      switch (command) {
+        case getBinaryData:
+          Map<String, byte[]> binaryResult =
+              ImmutableMap.of(ZooKeeperCommand.getBinaryData.name(), bytes);
+          // Note: this serialization (using ObjectMapper) will convert this byte[] into
+          // a Base64 String! The REST client (user) must convert the resulting String back into
+          // a byte[] using Base64.
+          return JSONRepresentation(binaryResult);
+        case getStringData:
+          Map<String, String> stringResult =
+              ImmutableMap.of(ZooKeeperCommand.getStringData.name(), new String(bytes));
+          return JSONRepresentation(stringResult);
+        default:
+          LOG.error("Unsupported command :" + command);
+          return badRequest("Unsupported command :" + command);
```

Comment:
Nit, space after the colon might be easier to read in the log, `command: abcd`?

Reply:
Added